src/main/java/org/matsim/episim/model/VaccinationByAge.java

```diff
@@ -16,27 +16,51 @@
 
 	private final SplittableRandom rnd;
 
+	private final int MAX_AGE = 130;
+
 	@Inject
 	public VaccinationByAge(SplittableRandom rnd) {
 		this.rnd = rnd;
 	}
 
 	@Override
 	public int handleVaccination(Map<Id<Person>, EpisimPerson> persons, boolean reVaccination, int availableVaccinations, int iteration, double now) {
+		List<EpisimPerson> candidates = new ArrayList<EpisimPerson>(persons.values().size());
+
+		// to improve the performance of sort below, we first remove the persons
```

Comment:
Now that I see your approach I got a different idea.
The persons don't really need to be sorted at all. Just put into bins with their age and vaccinated started at the oldest.
It would be a bit similar to a radix sort, but we don't necessarily require to have a sorted list.

Reply:
I have added a (untested) commit. Incl. a new feature, a minimum age for vaccinations, currently set to 6 years. In theory this changes the behavior compared to the previous version, but only for simulations where more then 90% get a vaccination.